src/queue.c

```diff
@@ -0,0 +1,101 @@
+#include "queue.h"
+
+#include <stdio.h>
+#include <stdlib.h>
+#define TRUE 1
+#define FALSE 0
+
+struct node
+{
+    long start_time;
+    long wait_time;
+    struct node *next;
+};
+typedef struct node node;
+
+struct queue
+{
+    int count;
+    node *front;
+    node *rear;
+};
+typedef struct queue queue;
```

Comment:
Let's avoid typedefs as it masks what object we are using (queue can be a struct, union, enum or base type etc). Let's stick to the C standard library convention of keeping structs as structs to ensure our code is readable and portable.